pkg/auth/authenticator.go

```diff
@@ -11,7 +12,8 @@ import (
 
 type Authenticator interface {
 	// Authenticate authenticates the user and returns an error if the authentication failed.
-	Authenticate() error
+	// Takes a context which can be used to interrupt the authentication.
+	Authenticate(ctx context.Context) error
```

Comment:
Hm. For backwards compatibility, it might be good to have a new interface `CancellableAuthenticator`. As this is a library, exported functions and interfaces should ideally not be changed if not needed.